File: backend/src/main/java/com/cesupa/backend/domain/patient/Patient.java
package com.cesupa.backend.domain.patient;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

import com.cesupa.backend.domain.Gender;

@Entity
@Table(name = "patient")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@EqualsAndHashCode
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String fullName;
    private LocalDate birthDate;

    @Column(unique = true)
    private String cpf;

    @Enumerated(EnumType.STRING)
    private Gender gender;

    private String phone;
    private String address;
    private String email;

    @Enumerated(EnumType.STRING)
    private PatientBloodType bloodType;

    private String knownAllergies;
    private Boolean active;
}

File: frontend/src/types/patient.ts
export interface Patient {
  id?: number;
  name: string;
  birthDate: string;
  cpf: string;
  gender: string;
  phone: string;
  address: string;
  email?: string;
  bloodType?: string;
  allergies?: string;
  status?: 'active' | 'inactive';
  createdAt?: string;
  updatedAt?: string;
}

export interface PatientFormData extends Omit<Patient, 'id' | 'createdAt' | 'updatedAt'> {} 